File: Arrays & strings/392. Is subsequence/Costra.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i  = 0;
        for (char ch : t){
            if (i == s.size()) return true;
            if (s[i] == ch) i++; 
        }
        if (i == s.size()) return true;

        return false;
    }
};

File: Arrays & strings/392. Is subsequence/Costra.ts
function isSubsequence(s: string, t: string): boolean {
    let i : number = 0;
    for (const ch of t){
        //  if (i == s.length) return true;
        if (s[i] == ch) i++; 
    }

    return (i == s.length);
};


//   Other solution

// function isSubsequence(s: string, t: string): boolean {
//     const res = t.split("").reduce<number>((acc, ch) => ch === s[acc] ? ++acc : acc,0);
//     return res === s.length;
// };
